aether-producer/producer/aether_producer.py

```
@@ -157,135 +288,255 @@ def get_entities(offset = None, max_size=1000):  # TODO implement batch pull by
-        self.kernel.refresh()
-        topics = self.kernel.Resource.Schema
-        for topic in topics:
-            self.streams[topic.name] = KafkaStream(topic.name, self.kernel)
-
+                self.logger.debug("Getting Changeset for %s" % self.name)
+                self.change_set = {}
+                new_rows = self.get_db_updates()
+                end_offset = new_rows[-1].get('modified')
+
+                bytes_writer = io.BytesIO()
+                writer = DataFileWriter(
+                    bytes_writer, DatumWriter(), self.schema, codec='deflate')
+
+                for row in new_rows:
+                    _id = row['id']
+                    msg = row.get("payload")
+                    self.change_set[_id] = row
+                    writer.append(msg)
+
+                writer.flush()
+                raw_bytes = bytes_writer.getvalue()
+                writer.close()
+
+                self.producer.produce(
+                    self.topic_name,
+                    raw_bytes,
+                    callback=self.kafka_callback
+                )
+                self.producer.flush()
+                self.wait_for_kafka(end_offset, failure_wait_time=self.kafka_failure_wait_time)
 
-    def send(self, row_generator):
-        for row in row_generator:
-            if self.killed: #look for sigterm
-                print ("manager stopped in progress via signal")
-                return
-            topic = row.get("schema_name")
-            self.streams[topic].send(row)
-        print ("manager finished processing changes")
-
-
-    def stop(self):
-        for name, stream in self.streams.items():
-            stream.stop()
-            print ("released connection to topic: %s" % name)
-        self.streams = {}
-        print ("manager stopped")
+            except Exception as ke:
+                self.logger.error('error in Kafka save: %s' % ke)
+                self.logger.error(traceback.format_exc())
+                sleep(1)
```

As I commented in another line, if we swallow the exception and silently log the error we will only realize that something is not working once we notice that the data is too old.

---

Something going wrong in this try block is likely due to systemic failure. I don't know how to scream more loudly than logger.error(). It would also be apparent from the /status endpoint that data is no longer being saved to Kafka. The other options would be killing the producer, but if it's a data issue causing this operation to fail, that would take down other topics for no good reason.